Lib/TLib.hs

```diff
@@ -266,12 +268,13 @@ withoutEach f (x:xs) = withoutEach' f ([],x,xs)
 --   This works by testing if the absence of any matching component of a player's message would cause a penalty containing 'failmsg'.
 --   If not, the component is assumed to be unnecessary.
 unnec :: String -> Rule
-unnec s act e@(Action p a m) gs =
+unnec s = unnec' s ("unnecessarily saying '{}'"%)
+unnec' s unnecmsg act e@(Action p a m) gs =
             let ms = split m
                 r = regexProcess s in
                 --lm = length $ _messages gs in
             foldr (.) id (withoutEach (\ms' x -> when (__$ isJust $ matchRegexAll r x)$
               when (__$not$ any (checkMatch$ "receives penalty.*{}"%failmsg x) (_messages$ act (Action p a (reconstitute ms')) gs))
```

Comment:
If `mustSay'` does not use the global `failmsg`, then this bit doesn't work. Worth being aware of.

Reply:
That's a good point, perhaps the most general version also needs a custom failmsg

Also, maybe the global failmsg should be called something like failmsgDefault to avoid confusion with using failmsg as a parameter name